### PR TITLE
MARS-249: fixed flash loading home page experiment

### PR DESCRIPTION
**Pull request by edwinf-barreram**
- Fixed issue while loading new home page experiment

# Before 

https://user-images.githubusercontent.com/102815442/193158512-5526e202-b4fc-4ed7-bf5e-134ff38dd1f9.mov

# Now

https://user-images.githubusercontent.com/102815442/193158590-f4f080a8-c9f7-4159-b1da-8685ed3b7ae8.mov

### CODE REVIEW

File: src/pages/ContentfulPage.vue [hunk 1]
```diff
@@ -265,10 +265,10 @@ export default {
 	},
 	apollo: {
 		query: contentfulEntries,
-		preFetchVariables({ route, client }) {
+		preFetchVariables({ route }) {
 			return {
 				contentType: 'page',
-				contentKey: route?.meta?.contentfulPage(route, client)?.trim(),
+				contentKey: route?.meta?.contentfulPage(route, this.apollo)?.trim(),
```
Comment (mcstover) [inline src/pages/ContentfulPage.vue:271]: Do your changes below work without changes to this file?

Comment (edwinf-barreram) [reply src/pages/ContentfulPage.vue:271]: It was the main problem, that client is undefined for some time.

Comment (mcstover) [reply src/pages/ContentfulPage.vue:271]: ok that's interesting

**Review by mcstover — APPROVED** [review]

File: src/pages/ContentfulPage.vue (hunk 1 above)
Comment (emuvente) [reply src/pages/ContentfulPage.vue:271]: `this` is also undefined in preFetchVariables during the prefetch process. Client needs to be passed in to the preFetchVariables function for this to render correctly on the server-side. I'm testing a fix for that now.

File: src/pages/ContentfulPage.vue [hunk 2]
```diff
@@ -305,7 +305,7 @@ export default {
 				query: contentfulEntries,
 				variables: {
 					contentType: 'page',
-					contentKey: args?.route?.meta?.contentfulPage(args?.route)?.trim(),
+					contentKey: args?.route?.meta?.contentfulPage(args?.route, client)?.trim(),
```
Comment (emuvente) [inline src/pages/ContentfulPage.vue:308]: This is what was needed to get the server-side render correct.

**Review by emuvente — APPROVED** [review]
👍 Looks good to go!

File: src/pages/ContentfulPage.vue (hunk 2 above)
Comment (edwinf-barreram) [reply src/pages/ContentfulPage.vue:308]: @emuvente Thank you very much!

**Review by mcstover — APPROVED** [review]

**Pull request merged**